src/main/java/org/spongepowered/api/data/property/PropertyHolder.java

```diff
@@ -24,38 +24,54 @@
  */
 package org.spongepowered.api.data.property;
 
-import org.spongepowered.api.data.Property;
-
-import java.util.Collection;
+import java.util.Map;
 import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
 
 public interface PropertyHolder {
 
     /**
-     * Attempts to retrieve a specific {@link Property} type of this
-     * {@link PropertyHolder}. If the property is not applicable,
-     * {@link Optional#empty()} is returned.
+     * Attempts to retrieve a value for the specified {@link Property}. If
+     * the property is not applicable, {@link Optional#empty()} is returned.
      *
      * <p>{@link Property}s can define various immutable information about a
      * {@link PropertyHolder} that is dependent on the instance of the holder.
      * As {@link Property}s cannot be changed, the {@link PropertyHolder} can
      * not change the information about it's own properties either.</p>
      *
-     * @param propertyClass The property class
-     * @param <T> The type of property
-     * @return The property, if available
+     * @param property The property to retrieve the value for
+     * @param <V> The property value type
+     * @return The property value, if available
+     */
+    <V> Optional<V> getProperty(Property<V> property);
```

Comment:
So the change with usage is to just query for the property value fo the holder. 👍 How does this interact with int properties that are returning the primitive equivalent?

Reply:
My guess is that primitives will be autoboxed into `Integer`s and then unboxed after `Optional.get()`.

Reply:
I added the following methods to get rid of boxing/unboxing:
```java
int getProperty(IntProperty property, int defaultValue);
double getProperty(DoubleProperty property, double defaultValue);
boolean getProperty(BooleanProperty property, boolean defaultValue);
```
I think that `int`, `double` and `boolean` would cover all the primitive properties types that are being used/will be used.

Reply:
String, byte, arrays?

Reply:
`String` isn't a primitive, so it doesn't really need boxing/unboxing, except for the optional. `byte`, `long`, `float[]`, etc. are never used by properties.

Reply:
You can also use OptionalInt and OptionalDouble. OptionalBoolean doesn’t exist (in all systems at least) but a Tristate may suffice.